app/src/main/java/com/duckduckgo/app/trackerdetection/model/TrackerNetworks.kt

```diff
@@ -53,11 +48,30 @@ class TrackerNetworksImpl @Inject constructor(val prevalenceStore: PrevalenceSto
         )
     }
 
+    private fun recursivelyFindMatchingTracker(url: String): DisconnectTracker? {
```

Comment:
I liked the name because normally calling the method you're currently inside is an error, but it's ok in this case only because it's recursive.

I could remove that naming though if it doesn't offer the safety feeling i intended it to

Reply:
It's an optional suggestion, so up to you. If you think people assume recursion is a mistake (I don't but perhaps most devs do?) then the naming convention to indicate intentionality makes sense.